app/models/import/technologies/fetcher.rb

```diff
@@ -0,0 +1,53 @@
+class Import
+  module Technologies
+    class Fetcher
+      def initialize(etm_scenario)
+        @etm_scenario = etm_scenario
+        @scenario_id  = etm_scenario.fetch(:id)
+      end
+
+      def fetch
+        builders.flat_map do |builder|
+          builder.new(gqueries, @etm_scenario).build(technologies_from)
+        end
+      end
+
+      private
+
+      # Internal: Imports the requested data from the remote provider and returns
+      # the JSON response as a Hash.
+      def response
+        @response ||= EtEngineConnector.new(keys: keys)
+          .stats(@scenario_id)['nodes']
+      end
+
+      def keys
+        Technology.importable.map(&:key)
+      end
+
+      # Internal: Given a response, splits out the nodes into discrete technologies.
+      #
+      # Returns an array.
+      def technologies_from
+        @technologies_from ||= response.flat_map do |key, data|
+          TechnologyBuilder.build(key, data)
+        end
+      end
+
+      def gqueries
```

Comment:
> I'm not sure where that extra request for "households_flexibility_p2p_electricity" comes from. Maybe the front-end form since "Battery" is the default selection in the "Add technology" field?

Yes. That's the main reason. 👍 I'll memoize it.